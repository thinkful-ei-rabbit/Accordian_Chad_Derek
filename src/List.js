import React, { Component } from 'react';
import Card from './Card'
import './styles/List.css';

class List extends Component {
  static defaultProps = {
    props: {
      key: '',
      header: '',
      cards: [],
    }
  };

  render() {
    const props = this.props;
    return (
      <section className='List'>
        <header className='List-header'>
          <h2>{props.header}</h2>
        </header>
        <div className='List-cards'>
          {props.cards.map((card) =>
            <Card
              key={card.id}
              title={card.title}
              content={card.content}
            />
          )}
          <button
            type='button'
            className='List-add-button'
          >
            + Add Random Card
          </button>
        </div>
      </section>
    )
  }
}

export default List