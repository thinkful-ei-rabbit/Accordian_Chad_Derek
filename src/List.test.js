import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

it('List renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List allCard={[]}/>, div);
    ReactDOM.unmountComponentAtNode(div);
});