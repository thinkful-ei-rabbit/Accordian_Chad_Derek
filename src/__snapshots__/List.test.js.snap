// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My List renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Header test 1
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
